Take config server out of rotation (status.html) until it is bootstra…

hmusum opened the pull request:
…pped

Also make sure that health API status code is 'initializing' until RPC
server is actually running.

Commit:
Take config server out of rotation (status.html) until it is bootstrapped

Also make sure that health API status code is 'initializing' until RPC
server is actually running.

hakonhall commented on configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java:
```diff
@@ -3,56 +3,70 @@
- * @author lulf
- * @since 5.1
+ * If config server has been upgraded to a new version since the last time it was running it will redeploy all
+ * applications. If that is done syúccessfully the RPC server will start and the health status code will change from
```
Fix "syúccessfully"

hakonhall commented on configserver/src/main/java/com/yahoo/vespa/config/server/ConfigServerBootstrap.java:
```diff
@@ -3,56 +3,70 @@
         this.stateMonitor = stateMonitor;
         this.serverThread = new Thread(this, "configserver main");
+        this.vipStatus = vipStatus;
+        initializing(); // Initially take server out of rotation
```
VipStatus.inRotation() is unfortunately true by default, so there will be a window where status.html says it's up and before we reach this point.

It would be nice to use a "DefaultOutOfRotationVipStatus" instead of VipStatus. Or add an initialInRotation (default true) to VipStatusConfig and have that injected to VipStatus? Third alternative is to use the file.

Commit:
Add initiallyInRotation to vip status config and inject config into VipStatus

Use config value for inital value of isInRotation() when nothing
is known about backend clusters
